testutil/pr/track.go

```diff
@@ -0,0 +1,362 @@
+// Copyright © 2022 Obol Labs Inc.
+//
+// This program is free software: you can redistribute it and/or modify it
+// under the terms of the GNU General Public License as published by the Free
+// Software Foundation, either version 3 of the License, or (at your option)
+// any later version.
+//
+// This program is distributed in the hope that it will be useful, but WITHOUT
+// ANY WARRANTY; without even the implied warranty of  MERCHANTABILITY or
+// FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+// more details.
+//
+// You should have received a copy of the GNU General Public License along with
+// this program.  If not, see <http://www.gnu.org/licenses/>.
+
+// Package pr provides functions to process GitHub pull requests.
+
+//nolint:wrapcheck,cyclop,exhaustruct,revive
+package pr
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"time"
+
+	gh "github.com/shurcooL/githubv4"
+	"golang.org/x/oauth2"
+
+	"github.com/obolnetwork/charon/app/errors"
+)
+
+const (
+	// Name of the GitHub organization.
+	organization = "ObolNetwork"
+	// The number of the project. For ex: https://github.com/orgs/ObolNetwork/projects/1 has projectNumber 1.
+	projectNumber = gh.Int(1)
+)
+
+// projectV2FieldValue is a value to use in updateProjectV2ItemFieldValueInput.
+// https://docs.github.com/en/graphql/reference/input-objects#projectv2fieldvalue
+type projectV2FieldValue struct {
+	Date                 gh.String `json:"date,omitempty"`
+	IterationID          gh.ID     `json:"iterationId,omitempty"`
+	Number               gh.Float  `json:"number,omitempty"`
+	SingleSelectOptionID gh.String `json:"singleSelectOptionId,omitempty"`
+	Text                 gh.String `json:"text,omitempty"`
+}
+
+// updateProjectV2ItemFieldValueInput is an input for UpdateProjectV2ItemFieldValue.
+// https://docs.github.com/en/graphql/reference/input-objects#updateprojectv2itemfieldvalueinput
+type updateProjectV2ItemFieldValueInput struct {
+	ClientMutationID gh.String           `json:"clientMutationId,omitempty"`
+	FieldID          gh.ID               `json:"fieldId"`
+	ItemID           gh.ID               `json:"itemId"`
+	ProjectID        gh.ID               `json:"projectId"`
+	Value            projectV2FieldValue `json:"value"`
+}
+
+// addProjectV2ItemByIdInput is an input for AddProjectV2ItemById.
+// https://docs.github.com/en/graphql/reference/input-objects#addprojectv2itembyidinput
+type addProjectV2ItemByIdInput struct { //nolint:revive,stylecheck
```

Comment:
can you use `gh.UpdatePullRequest`

Reply:
no. the same reason as below. The best replacement would have been `AddProjectNextItemInput` but it'll be deprecated